hieroplatform/src/main/java/org/hiero/sketches/HLogLog.java

```diff
@@ -0,0 +1,111 @@
+package org.hiero.sketches;
+
+import org.hiero.table.api.IColumn;
+import org.hiero.table.api.IMembershipSet;
+import org.hiero.table.api.IRowIterator;
+import org.hiero.utils.HashUtil;
+
+/**
+ * A class that computes an approximation of the number of distinct elements in a column. Elements
+ * are identified via their hashcode. The class uses the HyperLogLog algorithm for large estimates
+ * and LinearCounting algorithm for small estimates.
+ */
+public class HLogLog {
+    private final int regNum; //number of registers
+    private final int logregNum;
+    private final byte[] registers;
+    private final int seed;
+
+    /**
+     * @param logRegNum the logarithm of the number of registers. should be in 4...16
+     *        Going beyond 16 would make the data structure big for no good reason.
+     *        Setting logRegNum in at 10-12 should give roughly 2% accuracy most of the time
+     */
+    public HLogLog(int logRegNum, int seed) {
+        if (!spaceValid(logRegNum))
+            throw new IllegalArgumentException("HLogLog initialized with number out of range");
+        this.regNum = 1 << logRegNum;
+        this.registers = new byte[this.regNum];
+        this.logregNum = logRegNum;
+        this.seed = seed;
+    }
+
+    /**
+     * adds the int 'value' to the data structure.
+     * Uses the first bits to identify the register and then counts trailing zeros
+     * @param value already assumed to be a random hash of the item
+     */
+    private void add(int value) {
+        int index =  value >>> (Integer.SIZE - this.logregNum);
+        byte zeros = (byte) (Integer.numberOfTrailingZeros(value) + 1);
+        if (zeros > this.registers[index])
+            this.registers[index] = zeros;
+
+    }
+
+    /**
+     *
+     * Creates a Hyperloglog data structure from a column and membership set. Uses the hash code
+     * of the objects in the column as identifier.
+     */
+    public void createHLL(IColumn column, IMembershipSet memSet) {
+        final IRowIterator myIter = memSet.getIterator();
+        int currRow = myIter.getNextRow();
+        while (currRow >= 0) {
+            int value = column.getObject(currRow).hashCode();
```

Comment:
This will throw an exception if the object is missing.
Also, maybe you want to create a class HashCode which has a bunch of static methods for computing 64-bit hashcodes.
You may want to handle scalar specially, it you return an object it will perform boxing for every scalar, allocating an extra object.